### PR TITLE
Finilized Everything

### PR DESCRIPTION
**Pull request by ibugithub**
#In this branch I did the following
  - [ ] Added dependencies for the project
  - [ ] Setup rails
  - [ ] Implemented Authentications
 - [ ] Created a welcome page and a dashboard page
 - [ ] Implemented Categories page
- [ ] Added form for creating new category
- [ ] Implemented Transactions page
- [ ] Added form for creating new transaction
- [ ] Implemented unit testing's for all the controllers, models, views
- [ ] Deployed the production unit on render.com
- [ ] Added a presentation video of the app in the readme.

### CODE REVIEW

File: spec/controllers/expenses_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  include Devise::Test::ControllerHelpers
+  let(:user) { create(:user) }
+  let(:category) { create(:category, user:) }
+
+  before do
+    sign_in user
+  end
+
+  describe 'GET #new' do
+    it 'assigns a new expense to @expense' do
```
Comment (Olamarx) [inline spec/controllers/expenses_controller_spec.rb:13]: - [ ] I can see 21 of your test cases passed, good job but I will encourage you to double-check the remaining as they are failing as shown in the image below.
image

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @ibugithub,

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Linters passed ✔️
- Readme file professional ✔️
- Gitflow used ✔️
- Product deployed ✔️

## Required Changes ♻️
_Check the comments under the review._
- [ ] Good job recording the video but you forgot to point out the aspect of the code that interests you, kindly include it in the next one you will do.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/controllers/expenses_controller_spec.rb (hunk 1 above)
Comment (ibugithub) [reply spec/controllers/expenses_controller_spec.rb:13]: @Olamarx. Hello brother. I think you have some problems with your setup. Try to fix it. All of my tests are passing here in my system.
image


Check out the video. All of my tests have been passed. 
https://www.loom.com/share/18a2fb96091946cf9d6036ac5a29d242

**Review by mikethreels — APPROVED** [review]
Hi Ibu, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- good job on making the project work correctly
## Optional suggestions
- [ ] [OPTIONAL] Because the previous CR didn't mention this it will not be marked as a required change, however, according to the requirements You should follow these [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), including:
    Colors.
    Typography: font face, size and weight.
    Layout: composition and space between elements.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**